rts/System/SpringApp.cpp

```diff
@@ -538,6 +539,20 @@ void SpringApp::ParseCmdLine(int argc, char* argv[])
 		AILibraryManager::OutputSkirmishAIInfo();
 		exit(spring::EXIT_CODE_SUCCESS);
 	}
+	else if (!FLAGS_calc_checksum.empty()) {
+		ConsolePrintInitialize(FLAGS_config, FLAGS_safemode);
+		FileSystemInitializer::InitializeTry();
+		archiveScanner->ResetNumFilesHashed();
+		const std::string archive = archiveScanner->ArchiveFromName(FLAGS_calc_checksum);
+		const auto cs = archiveScanner->GetArchiveCompleteChecksumBytes(archive);
+
+		sha512::hex_digest hexCs = {0};
+		sha512::dump_digest(cs, hexCs);
+
+		LOG("Archive \"%s\", checksum = \"%s\"", FLAGS_calc_checksum.c_str(), hexCs.data());
```

Comment:
Is this the only output of the exe with the flag? If so, it would be good if it was structured somehow for easy parsing (json? `%s\n%s\n`? idk). If it actually adds an entry to archive cache then disregard this.

Reply:
It does add entries to the `ArchiveCache%d.lua`, the log is mostly for posterity.